sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/clangtidy/CxxClangTidySensor.java

```diff
@@ -0,0 +1,125 @@
+
+  private class Issue {
+    String path;
+    String line;
+    String level;
+    String message;
+    String check;
+  }
+  
+  // Extended issue information disabled for now as this has some formatting issues in the UI (not very readable and ugly).
+  private boolean extendedIssueInformation = false;
+
+  /**
+   * {@inheritDoc}
+   */
+  public CxxClangTidySensor(Settings settings) {
+    super(settings, CxxMetrics.CLANGTIDY);
+  }
+
+  @Override
+  protected String reportPathKey() {
+    return REPORT_PATH_KEY;
+  }
+
+  @Override
+  public void describe(SensorDescriptor descriptor) {
+    descriptor.onlyOnLanguage(CxxLanguage.KEY).name("CxxClangTidySensor");
+  }
+
+  @Override
+  protected void processReport(final SensorContext context, File report) {
+    LOG.debug("Parsing clang-tidy report");
+
+    try (Scanner scanner = new Scanner(report, "UTF-8")) {
+      // E:\Development\SonarQube\cxx\sonar-cxx\sonar-cxx-plugin\src\test\resources\org\sonar\plugins\cxx\reports-project\clang-tidy-reports\..\..\cpd.cc:76:20: warning: ISO C++11 does not allow conversion from string literal to 'char *' [clang-diagnostic-writable-strings]
+      // case 1: return "one";
+
+      // <path>:<line>:<column>: <level>: <message> [<checkname>]
+      Pattern p = Pattern.compile("([^:]+):([0-9]+):([0-9]+): ([^:]+): ([^]]+) \\[([^]]+)\\]");
+      Pattern p_windows = Pattern.compile("([^:]+):([^:]+):([0-9]+):([0-9]+): ([^:]+): ([^]]+) \\[([^]]+)\\]");
+      Issue currentIssue = null;
+
+      while (scanner.hasNextLine()) {
+        String line = scanner.nextLine();
+        Matcher matcher = p.matcher(line);
+        Matcher matcherWindows = p_windows.matcher(line);
+        if (matcher.matches()) {
+          if (currentIssue != null) {
+            saveUniqueViolation(context, CxxClangTidyRuleRepository.KEY, currentIssue.path, currentIssue.line, currentIssue.check,
+                currentIssue.message);
+          }
+          MatchResult m = matcher.toMatchResult();
+          currentIssue = new Issue();
+          currentIssue.path = m.group(1);
+          currentIssue.line = m.group(2);
+          currentIssue.level = m.group(4);
+          currentIssue.message = m.group(5);
+          currentIssue.check = m.group(6);
+        } else if(matcherWindows.matches()) {
+          if (currentIssue != null) {
+            saveUniqueViolation(context, CxxClangTidyRuleRepository.KEY, currentIssue.path, currentIssue.line, currentIssue.check,
+                currentIssue.message);
+          }
+          MatchResult m = matcherWindows.toMatchResult();
+          currentIssue = new Issue();
+          currentIssue.path = m.group(1) + ":" + m.group(2);
+          currentIssue.line = m.group(3);
+          currentIssue.level = m.group(5);
+          currentIssue.message = m.group(6);
+          currentIssue.check = m.group(7);          
+        } else if (extendedIssueInformation && currentIssue != null) {
```

Comment:
not sure @bjd what was the purpose of this, do you have  a example report that would create this?